Bio/SeqFeature.py

```diff
@@ -669,10 +669,10 @@ def __init__(self, start, end, strand=None, ref=None, ref_db=None):
             self._end = ExactPosition(end)
         else:
             raise TypeError("end=%r %s" % (end, type(end)))
-        if self._start > self._end:
+        if self.start.position and self.end.position and self.start > self.end:
```

Comment:
Sadly zero is a valid start or end position, so I think this would miss some corner cases